no-whitespace-for-layout false positives

```hbs
<div
  foo="bar"
  baz="qux"
>
  example
</div>

{{#foo-bar}}
  {{#baz-qux foo="bar"}}example{{/baz-qux}}
  {{#baz-qux foo="bar"}}example{{/baz-qux}}
{{/foo-bar}}

```

Actual result:

```text
  4:1  error  Excess whitespace detected.  no-whitespace-for-layout
  8:12  error  Excess whitespace detected.  no-whitespace-for-layout
  9:43  error  Excess whitespace detected.  no-whitespace-for-layout
```

Expected result: no warnings

--- a/lib/rules/no-whitespace-for-layout.js
+++ b/lib/rules/no-whitespace-for-layout.js
@@ -8,15 +8,24 @@ module.exports = class NoWhitespaceForLayout extends Rule {
     return {
       TextNode(node) {
         let source = this.sourceForNode(node);
-        // check for two ` ` or `&nbsp;` in a row
-        let matches = source.match(/(( )|(&nbsp;))(( )|(&nbsp;))/g);
-        if (matches !== null) {
-          this.log({
-            message: ERROR_MESSAGE,
-            line: node.loc && node.loc.start.line,
-            column: node.loc && node.loc.start.column,
-            source: this.sourceForNode(node),
-          });
+
+        let lines = source.split('\n');
+        for (let line of lines) {
+          // ignore whitespace at the start and end of the line
+          let trimmed = line.trim();
+
+          // check for two ` ` or `&nbsp;` in a row
+          let matches = trimmed.match(/(( )|(&nbsp;))(( )|(&nbsp;))/g);
+          if (matches !== null) {
+            this.log({
+              message: ERROR_MESSAGE,
+              line: node.loc && node.loc.start.line,
+              column: node.loc && node.loc.start.column,
+              source: this.sourceForNode(node),
+            });
+
+            return;
+          }
         }
       },
     };

--- a/test/unit/rules/no-whitespace-for-layout-test.js
+++ b/test/unit/rules/no-whitespace-for-layout-test.js
@@ -9,7 +9,13 @@ generateRuleTests({
   name: 'no-whitespace-for-layout',
   config: true,
 
-  good: ['Start to finish', 'Start&nbsp;to&nbsp;finish', 'Start<br>to<br>finish'],
+  good: [
+    'Start to finish',
+    'Start&nbsp;to&nbsp;finish',
+    'Start<br>to<br>finish',
+    '<div>\n  example\n</div>',
+    '<div\n  foo="bar"\n  baz="qux"\n>\n  example\n</div>',
+  ],
 
   bad: [
     {

